torch/testing/_internal/common_methods_invocations.py

```diff
@@ -280,6 +311,60 @@ def __init__(self,
         #   outside a function's domain.
         self._domain_eps = 1e-5
 
+    def sample_sparse_inputs(self, device, dtype, requires_grad=False, **kwargs):
+        """Returns an iterable of SampleInputs."""
+        if not self.supports_sparse:
+            raise RuntimeError("This op does not support sparse input")
+
+        low, high = self.domain
+        low = -10 if low is None else low + self._domain_eps
+        high = 10 if high is None else high - self._domain_eps
+        is_coalesced = kwargs.get("is_coalesced", False)
+
+        numel = 10
+        t1 = torch.sparse_coo_tensor(
+            indices=torch.arange(numel).unsqueeze(0),
+            values=torch.linspace(low, high, steps=numel),
+            size=[numel, ],
+            device=device,
+            dtype=dtype,
+        )
+        # hybrid sparse input
+        size = [10, 12, 4]
+        numel = 8
+        i1 = torch.randint(0, size[0], size=(numel, ))
+        i2 = torch.randint(0, size[1], size=(numel, ))
+        indices = torch.stack([i1, i2], dim=0)
+        values = torch.linspace(low, high, steps=numel * size[2]).reshape(numel, -1)
+        t2 = torch.sparse_coo_tensor(
+            indices=indices,
+            values=values,
+            size=size,
+            device=device,
+            dtype=dtype
+        )
+
+        # empty sparse tensor
+        t3 = torch.sparse_coo_tensor(
+            indices=torch.zeros([2, 0]),
+            values=torch.zeros([0, 5, 5, 5, 5, 5, 5, 0]),
+            size=[0, 0, 5, 5, 5, 5, 5, 5, 0],
+            device=device,
+            dtype=dtype
+        )
+
+        if is_coalesced:
```

Comment:
See above comment. Let's always generate all the relevant sample inputs (and document which are generated in the docstring).